### PR TITLE
[SDPA-4591] adds tide_site custom path processor

### PR DESCRIPTION
**Pull request by vincent-gao**
### JIRA
https://digital-engagement.atlassian.net/browse/SDPA-4591

### Issue
This is a common issue when using multiple urls. by default, you hit the one of urls that attached to a node, it will take you to the last revision of the path. 
eg. node/123 has `/site-1/hello-world`  and `/site-2/hello-world` path aliases.
you hit `/site-1/hello-world` in the browser, it will redirect you to `/site-2/hello-world` as the later is newest one.

### Changes
This is a custom function that redirects the node url to primary site based path alias.

### CODE REVIEW

**Review by GROwen — CHANGES_REQUESTED** [review]
@vincent-gao I've put an update on the Jira issue that explains why some further investigation is required

File: src/PathProcessor/TideSitePathProcessor.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    if (empty($options['alias'])) {
+      $langcode = isset($options['language']) ? $options['language']->getId() : NULL;
+      $pattern = "\/node\/[0-9]*";
+      // We only care about /node/{id} path.
+      if (preg_match('/^' . $pattern . '$/', $path)) {
+        // If it is an node canonical url, we load it.
+        if (preg_match("/\/(\d+)$/", $path, $matches)) {
+          $nid = $matches[1];
+          $node = Node::load($nid);
+          $aliases = $this->tidAliasHelper->loadAll(['path' => $path]);
```
[the remark this answers is not in the document]
Comment (vincent-gao) [reply src/PathProcessor/TideSitePathProcessor.php:79]: Thanks @GROwen ! thanks about this, will change.

File: src/PathProcessor/TideSitePathProcessor.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  public function __construct(AliasManagerInterface $alias_manager, TideSiteHelper $siteHelper, AliasStorageHelper $aliasStorageHelper) {
+    $this->aliasManager = $alias_manager;
+    $this->tideSiteHelper = $siteHelper;
+    $this->tideAliasHelper = $aliasStorageHelper;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processInbound($path, Request $request) {
```
Comment (anthony-malkoun) [inline src/PathProcessor/TideSitePathProcessor.php:61]: @vincent-gao is this required by one of the interfaces? Is it doing anything different than the default implementation?

Comment (vincent-gao) [reply src/PathProcessor/TideSitePathProcessor.php]: @anthony-malkoun , No , it doesn't required. deleted. Thanks for your CR.

**Pull request closed without merging**